thetis/turbulence.py

```diff
@@ -1077,3 +1077,146 @@ def __init__(self, function_space, gls_model,
                                bathymetry, v_elem_size, h_elem_size)
         self.add_term(diff, 'implicit')
         self.add_term(source, 'implicit')
+
+
+class PacanowskiPhilanderOptions(AttrDict):
+    """
+    Options for Generic Length Scale turbulence model
+    """
+    def __init__(self):
+        super(PacanowskiPhilanderOptions, self).__init__()
+        self.max_viscosity = 5e-2
+        r"""float: Constant maximum viscosity :math:`\nu_{max}`"""
+        self.alpha = 10.0
+        """float: Richardson number multiplier"""
+        self.exponent = 2.0
+        """float: Exponent of viscosity numerator :math:`n`"""
+
+    def print_summary(self):
+        """Prints all defined parameters and their values."""
+        print_output('Pacanowski-Philander model parameters')
+        for k in sorted(self.keys()):
+            print_output('  {:16s} : {:}'.format(k, self[k]))
+
+
+class PacanowskiPhilanderModel(object):
```

Comment:
Should we maybe have a base TurbulenceModel class? So we can document what the expected API is?